pkg/innerring/processors/container/common.go

```diff
@@ -79,10 +79,6 @@ func (cp *Processor) verifySignature(v signatureVerificationData) error {
 			return errors.New("session token is not signed by the container owner")
 		}
 
-		if !tok.AssertAuthKey(&key) {
```

Comment:
Iinm Chuck could attach the token Alice issued to Bob along with his own signature to mutate Alice's container. This condition prevents such a malicious behavior. Dropping the condition leaves [`session_key`](https://github.com/nspcc-dev/neofs-api/blob/33e3a77997674e576070db452f84583576a86368/session/types.proto#L112-L113) field unhandled

Reply:
https://github.com/nspcc-dev/neofs-sdk-go/blob/ac5d07bb6540e4f1f097a28e21ee20e1381f31ba/session/container.go#L246 uses `authKey` (which is a `session_key`) to verify provided signature. So the signature has to be created using the key specified in the token.

Reply:
alright. Then Chuck cannot change the request payload, so he can only fake the fact that Bob _sent_ the transaction when Bob wasn't going to. I see it as a special case of a replay attack. This check does not protect anyway, so lets drop